lib/manageiq/api/client/authentication.rb

```diff
@@ -25,6 +25,10 @@ def inspect
           super.gsub(/@password=".+?"(, )?/, "")
         end
 
+        def self.options_specified?(options)
```

Comment:
Still don't like this name.  I read `options_specified?` and to me that says "Does this options hash have anything in it?", not "Does this options hash have the necessary authentication keys?".  Perhaps `auth_options_specified?` or `valid_auth_options?`?